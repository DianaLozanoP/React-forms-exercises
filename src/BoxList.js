import { useState } from "react"
import { v4 as uuid } from "uuid"
import Box from "./Box"
import NewBoxForm from "./NewBoxForm"

const BoxList = () => {
    const INITIAL_STATE = [
        { id: uuid(), backgroundColor: "pink", width: "80px", height: "80px" }
    ]
    const [boxes, setBoxes] = useState(INITIAL_STATE)
    const addBox = ({ backgroundColor, width, height }) => {
        setBoxes(boxes => [...boxes, { id: uuid(), backgroundColor, width, height }])
    }
    const removeBox = id => {
        const newBoxes = boxes.filter(box => box.id !== id)
        setBoxes(newBoxes)
    }
    return (
        <div>
            <h3>Boxes List</h3>
            <div>
                <ul>
                    {boxes.map(box => <Box
                        key={box.id}
                        id={box.id}
                        backgroundColor={box.backgroundColor}
                        width={box.width}
                        height={box.height}
                        removeBox={removeBox} />)}
                </ul>
            </div>
            <NewBoxForm addBox={addBox} />
        </div>
    )
}

export default BoxList;