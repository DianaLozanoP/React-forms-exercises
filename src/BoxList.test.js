import { fireEvent, render } from '@testing-library/react'
import BoxList from './BoxList'

test("it renders without crashing", () => {
    render(<BoxList />)
})

test("matches snapshot", () => {
    const { asFragment } = render(<BoxList />)
    expect(asFragment()).toMatchSnapshot();
})

test("should add new box", () => {
    const { getByLabelText, container, getByText } = render(<BoxList />)
    const inputColor = getByLabelText("Background Color")
    const inputWidth = getByLabelText("Width")
    const inputHeight = getByLabelText("Height")
    const btn = getByText("Add Box")
    const ul = container.querySelector('ul')
    fireEvent.change(inputColor, { target: { value: "Purple" } })
    fireEvent.change(inputWidth, { target: { value: "200px" } })
    fireEvent.change(inputHeight, { target: { value: "400px" } })
    fireEvent.click(btn);
    expect(ul).toContainHTML(`<ul><div style="background-color: pink; width: 80px; height: 80px; position: relative; margin-bottom: 10px;"><button style="position: absolute; top: 0px; right: -25px;">X</button></div><div style="background-color: Purple; width: 200px; height: 400px; position: relative; margin-bottom: 10px;"><button style="position: absolute; top: 0px; right: -25px;">X</button></div></ul>`)
})