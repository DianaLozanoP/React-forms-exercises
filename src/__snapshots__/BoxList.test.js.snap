// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <h3>
      Boxes List
    </h3>
    <div>
      <ul>
        <div
          style="background-color: pink; width: 80px; height: 80px; position: relative; margin-bottom: 10px;"
        >
          <button
            style="position: absolute; top: 0px; right: -25px;"
          >
            X
          </button>
        </div>
      </ul>
    </div>
    <form>
      <label
        for="backgroundColor"
      >
        Background Color
      </label>
      <input
        id="backgroundColor"
        name="backgroundColor"
        placeholder="Pink"
        type="text"
        value=""
      />
      <label
        for="width"
      >
        Width
      </label>
      <input
        id="width"
        name="width"
        placeholder="100px"
        type="text"
        value=""
      />
      <label
        for="height"
      >
        Height
      </label>
      <input
        id="height"
        name="height"
        placeholder="100px"
        type="text"
        value=""
      />
      <button>
        Add Box
      </button>
    </form>
  </div>
</DocumentFragment>
`;
